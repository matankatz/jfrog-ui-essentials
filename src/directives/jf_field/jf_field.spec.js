'use strict';
describe('unit test:jf_field directive', function () {
    var element,
        inputElement,
        getErrorElement,
        buttonElement,
        $timeout,
        $scope,
        JFrogEventBus;

    function setup(_JFrogEventBus_, _$timeout_) {
        JFrogEventBus = _JFrogEventBus_;
        $timeout = _$timeout_;
    }

    function compileDirective() {
        $scope = compileHtml('<form novalidate><jf-field validations="test"><input required name="name" minlength="0" ng-model="name"/></jf-field><button></button></form>');
        $scope.$digest();
        element = $(document.body).find('jf-field');
        inputElement = element.find('input');
        buttonElement = $(document.body).find('form');
        getErrorElement = () => $(document.body).find('[ng-message-exp]');
    }

    beforeEach(angular.mock.module('jfrog.ui.essentials'));
    beforeEach(inject(setup));
    beforeEach(compileDirective);

    it('should compile', () => {
        expect(element[0]).toMatchSnapshot();
    });

    it('should focus', () => {
        $scope = compileHtml('<form novalidate><jf-field autofocus="true" validations="test"><input required name="name" minlength="0" ng-model="name"/></jf-field><button></button></form>');
        $scope.$digest();
        element = $(document.body).find('jf-field');

        expect(element[0]).toMatchSnapshot();
    });

    it('should show elements', function () {
        expect(element.length).not.toEqual(0);
        expect(buttonElement.length).not.toEqual(0);
        expect(inputElement.length).not.toEqual(0);
    });

    it('should not show error', function () {
        expect(getErrorElement().length).toEqual(0);
    });

    it('should show error after submitting', function () {
        $(inputElement).trigger('blur');
        $scope.$digest();
        $timeout.flush();
        expect(getErrorElement().length).not.toEqual(0);
    });
});
