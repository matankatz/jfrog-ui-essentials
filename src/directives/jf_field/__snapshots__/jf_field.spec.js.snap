// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit test:jf_field directive should compile 1`] = `
<jf-field
  class="ng-isolate-scope"
  validations="test"
>
  <div
    ng-class="{'sticky-errors': dontPushDownErrors}"
  >
    
    
    <div
      class="jf-field"
      ng-class="{'input-label': animated}"
    >
      
        
      <ng-transclude>
        <input
          class="ng-pristine ng-untouched ng-scope ng-empty ng-invalid ng-invalid-required ng-valid-minlength"
          minlength="0"
          name="name"
          ng-model="name"
          required=""
        />
      </ng-transclude>
      
    
    </div>
    
    
    <!-- ngIf: (formField.showErrors || alwaysShowErrors || (formField.initialValue && !formField.$error.required)) && !formField.isAutoFocused -->
    

  </div>
  

</jf-field>
`;

exports[`unit test:jf_field directive should focus 1`] = `
<jf-field
  autofocus="true"
  class="ng-isolate-scope"
  validations="test"
>
  <div
    ng-class="{'sticky-errors': dontPushDownErrors}"
  >
    
    
    <div
      class="jf-field"
      ng-class="{'input-label': animated}"
    >
      
        
      <ng-transclude>
        <input
          class="ng-pristine ng-untouched ng-scope ng-empty ng-invalid ng-invalid-required ng-valid-minlength"
          minlength="0"
          name="name"
          ng-model="name"
          required=""
        />
      </ng-transclude>
      
    
    </div>
    
    
    <!-- ngIf: (formField.showErrors || alwaysShowErrors || (formField.initialValue && !formField.$error.required)) && !formField.isAutoFocused -->
    

  </div>
  

</jf-field>
`;
